dask/tests/test_base.py

```diff
@@ -1344,7 +1344,7 @@ def test_num_workers_config(scheduler):
     f = delayed(pure=False)(time.sleep)
     # Be generous with the initial sleep times, as process have been observed
     # to take >0.5s to spin up
-    a = [f(1.0), f(1.0), f(1.0), f(0.1)]
+    a = [f(2.0), f(2.0), f(2.0), f(0.1)]
```

Comment:
This test passes for me locally as it was, but on CI it seems these are not quite long enough as [CI counts 2 workers instead of 3]( https://github.com/dask/dask/runs/2157366345?check_suite_focus=true#step:7:17909 ). Apparently this is a known issue as the test is marked as flaky and there is a comment about this above as well. Trying to bump the `sleep`s a bit more to see if that will fix the test

Reply:
Ah scratch that. Appears to be a consequence of how batching works.

To elaborate, batching will result in combining 2 tasks per ever 1 batched task. So we will only see 2 workers used then. This is done to make better use of threads and processes as well as cutdown on task submission overhead (paying it once per batch instead of once per individual task).

To fix this, we just need the same # of tasks as workers. This way batching will just result in 1 task per batch. Thus there will be 3 batched tasks with 1 per worker.

Pushed a fix to change this in the most recent commit.

Reply:
Also turns out this wasn't actually get run locally as it was marked `slow`, which caused it to be skipped. After ensuring it did run locally can confirm that using the same # of tasks as workers fixes the issue